### PR TITLE
Utilize newTotalSupply field from HAPI

### LINKED ISSUE #1397: Support newTotalSupply
**Problem**
HAPI[ 0.10.0 ](https://docs.hedera.com/guides/docs/release-notes/services#v-0-10-0)added a `newTotalSupply` field to the transaction receipt for `TokenMint` and `TokenBurn`.

**Solution**
We should support the new field (as we're the ones who asked for it) and remove our code that looks up old supply. Also confirm nothing else in proto changed.

**Alternatives**

**Additional Context**
See service PR providing feature - https://github.com/hashgraph/hedera-services/pull/845

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
HAPI [0.10.0](https://docs.hedera.com/guides/docs/release-notes/services#v-0-10-0) added a newTotalSupply field in Services PR ( https://github.com/hashgraph/hedera-services/pull/845) to the transaction receipt for TokenMint and TokenBurn.

- Update pom.xml to use `hedera-protobuf.version` 0.10.0
- Update `EntityRecordItemListener` to pull `newTotalSupply` from recordItem and update token domain object TotalSupply with value
- Update `TokenRepository` with `updateTokenSupply` method that sets `totalSupply` and `modifiedTimestamp` based on `tokenId`
- Fix associated tests

**Which issue(s) this PR fixes**:
Fixes #1397 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java [hunk 1]
```diff
@@ -478,18 +478,11 @@ private void insertTokenAssociate(RecordItem recordItem) {
...
-            Optional<Token> optionalToken = retrieveToken(tokenID, TransactionTypeEnum.TOKENBURN, consensusTimeStamp);
-            if (optionalToken.isPresent()) {
-                Token token = optionalToken.get();
-                token.setModifiedTimestamp(consensusTimeStamp);
-                // mirror will calculate new totalSupply as an interim solution until network returns it
-                token.setTotalSupply(token.getTotalSupply() - tokenBurnTransactionBody.getAmount());
-
-                entityListener.onToken(token);
-            }
+            updateTokenSupply(
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java:482]: What happens if the token create occurs in the same record file?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -62,7 +62,7 @@
         <embedded.testcontainers.version>1.86</embedded.testcontainers.version>
         <grpc-spring-boot.version>2.10.1.RELEASE</grpc-spring-boot.version>
         <grpc.version>1.33.1</grpc.version>
-        <hedera-protobuf.version>0.9.0</hedera-protobuf.version>
+        <hedera-protobuf.version>0.10.0</hedera-protobuf.version>
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply pom.xml:65]: Yeah other than `newTotalSupply` I didn't find anything that affect mirror node in terms of response code, transaction body, record or receipt changes.
Lots of good comments though

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java:482]: Good point, that was the benefit of the previous find and update logic.
We could explore utilizing the cache in `SqlEntityListener` like we do with  `entityIds`

Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java:482]: Turns out we immediately saved on Token create so we're covered here.
However, will add a test to cover create and say wipe in the same record file.

Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java:482]: Added test to verify assumed coverage

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/repository/TokenRepositoryTest.java [hunk 3]
```diff
@@ -28,20 +30,45 @@ void save() throws DecoderException {
...
+    }
+
+    @Test
+    void updateSupplyOnMissingToken() throws DecoderException {
+        long createdTimestamp = 1L;
+        Token token = tokenRepository.save(token(createdTimestamp));
+        long newTotalSupply = INITIAL_SUPPLY - 1000;
+        long modifiedTimestamp = 5L;
+        tokenRepository.updateTokenSupply(new Token.Id(EntityId
+                .of("0.0.555", EntityTypeEnum.TOKEN)), newTotalSupply, modifiedTimestamp);
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/repository/TokenRepositoryTest.java]: Updated

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/repository/TokenRepositoryTest.java [hunk 4]
```diff
@@ -28,20 +30,45 @@ void save() throws DecoderException {
                 .get()).isEqualTo(token);
     }
 
+    @Test
+    void updateSupply() throws DecoderException {
+        Token token = tokenRepository.save(token(1));
+        long newTotalSupply = INITIAL_SUPPLY - 1000;
+        long modifiedTimestamp = 5L;
+        tokenRepository.updateTokenSupply(new Token.Id(FOO_COIN_ID), newTotalSupply, modifiedTimestamp);
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/repository/TokenRepositoryTest.java]: Updated

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**